### PR TITLE
Preserve book and label data , Add schema file

### LINKED ISSUE #15: [1.5pt] Preserve Book data in JSON file
All data should be preserved by saving collections in .json files.

### LINKED ISSUE #18: [1pt] Create schema file for Book and Label
Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
- books table (add all properties and associations from the parent Item class as table columns)
- labels table

### PR DESCRIPTION
**Pull request by asharanjith**
In this PR

- [x] Add storage for the books and labels in JSON files
- [x] Add schema of book, label and item class

### CODE REVIEW

File: database/schema.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+);
+
+CREATE TABLE book (
+  id INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY ,
+  publisher VARCHAR(255) NOT NULL,
+  cover_state VARCHAR(25) NOT NULL,  
+  CONSTRAINT fk_item FOREIGN KEY (item_id) REFERENCES item(id) ON DELETE CASCADE
+);
+
+CREATE TABLE label (
```
Comment (Tamunokuro) [inline database/schema.sql]: image

Comment (Tamunokuro) [reply database/schema.sql]: Hello Asha, I get the following errors when I run the schema codes.

**Review by Tamunokuro — CHANGES_REQUESTED** [review]
# REQUESTED CHANGES ♻️♻️

### Hi @asharanjith 

Great job on the work done so far but there are some issues. You are almost there!
 
almost there

## Highlights
- Professional PR :heavy_check_mark: 
- No linter errors :heavy_check_mark: 
- Good use of git-flows  :heavy_check_mark:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+ALTER TABLE book ADD CONSTRAINT fk_item FOREIGN KEY (id) REFERENCES item(id) ON DELETE CASCADE;
+
+ALTER TABLE label ADD CONSTRAINT fk_item FOREIGN KEY (id) REFERENCES book(id) ON DELETE CASCADE;
+
+ALTER TABLE item ADD CONSTRAINT fk_item FOREIGN KEY (id) REFERENCES book(id) ON DELETE CASCADE;
+
+ALTER TABLE item ADD CONSTRAINT fk_item FOREIGN KEY (id) REFERENCES game(id) ON DELETE CASCADE;
+
+ALTER TABLE item ADD CONSTRAINT fk_item FOREIGN KEY (id) REFERENCES author(id) ON DELETE CASCADE;
```
Comment (Tamunokuro) [inline database/schema.sql:41]: image

I get the above errors when I run the codes
I think you can use ALTER TABLE book ADD FOREIGN KEY (id) REFERENCES item(id) ON DELETE CASCADE for the first query and then replicate it for the others

**Review by Tamunokuro — CHANGES_REQUESTED** [review]
# REQUESTED CHANGES ♻️♻️

### Hi @asharanjith 

Great job on the work done so far but there are some issues. You are almost there!
 
almost there

## Highlights
- Professional PR :heavy_check_mark: 
- No linter errors :heavy_check_mark: 
- Good use of git-flows  :heavy_check_mark:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: database/schema.sql (hunk 2 above)
Comment (asharanjith) [reply database/schema.sql:41]: Hi @Tamunokuro 

Thanks for reviewing the PR

book
item
label


The table is generated fine. Please join a call if needed

**Review by Tamunokuro — APPROVED** [review]
# STATUS APPROVED 🟢🟢

### Hi @asharanjith 

> Good job! :+1: 
Your feature is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉 

approved


## Highlights
- Professional PR :heavy_check_mark: 
- No linter errors :heavy_check_mark: 
- Good use of git-flows  :heavy_check_mark:

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**